Zemnmez/monorepo
# airlock loadout

- Initial commit
- Fix the CI
- Return the nightly flags as 1.56 is still not around
- Fix a typo
- Rename GitHub action Rust to CI
- Put a CI badge
- Put a CI badge
- Put a CI badge
- #1
- Small rewording
- Remove the builder parameter as it is not necessary
- #4 - no_std support
- Fix formatting
- Fixes to the CI
- Fix formatting
- Fix formatting
- Fix formatting
- Try CI on 4.4 instead of 5.0
- Use latest esp-idf-sys to fix no_std builds on RiscV; take cfg parameters
- Fix the no_std build for RiscV32
- Fix no_std build
- Wrong reference to v4.4 branch
- Wrong reference to v4.4 branch
- Try the master branch
- CMake-first builds; documentation for cargo-pio (#3)
- CMake-first builds; documentation for cargo-pio (#3)
- Fix incorrect req about Python version
- Put a comment describing other ESP-IDf components the user can enable
- Small addition to the idf.py build step
- Fix the initial version in Cargo.toml
- Fix the CI
- CMake CI
- Put size optimizations
- Fix a broken dependency fromt he lib to the project
- Main task stack size of 7K
- Update to lateast dependencies
- Fix the settings for release vs debug build
- New release due to embedded-svc 0.13
- Remove `allow` declaration in favour of anonymous import
- Remove `allow` declaration in favour of anonymous import in cmake template
- Fix formatting
- Fix formatting
- Raise dependency to latest embedded-svc; fix obsolete references to the ivmarkov repo
- Use latest no_std-compatible embedded-svc
- Upgrade to latest esp-idf-sys
- Remove the single-component esp-idf-svc import because clippy complains
- Update all dependencies
- Update with mqtt component
- Small wording fix
- Update Cargo.toml
- Update Cargo.toml
- Clarify how to invoke the native build feature
- Update to latest dependencies; isolate ESP-IDF-master build in a separate workflow
- Update Cargo.toml
- Update Cargo.toml
- Make the CMake build compatible with latest esp-idf-sys
- Update to latest dependencies
- Fix the CIbuild
- add CONFIG_FREERTOS_HZ=1000 to sdkconfig.defaults
- Comment out CONFIG_FREERTOS_HZ=1000
- #12 and update to latest dependencies
- Update to latest dependencies
- Update with latest dependencies; fix the CI
- Default to v4.4, removing panic_immediate_abort feature when targetting v4.4+
- Switch back to latest nightly; raise esp-idf-svc dependency
- Fix CI
- Rename v5.0, to mainline and add an indicator that it's unstable.
- Document default builder behavior
- Update to latest dependencies
- CMake build fixes
- Check if driver is optional component in esp idf 5
- Add devcontainers for cargo
- Move .vscode and format devcontainer.json
- Update archs
- Update format
- Update architectures and formatting
- Update riscv gdb path
- Use binaries for wokwi-server and web-flash
- Remove zip file
- Add readme
- Rename file
- Move readme to docs
- Add template readme for cargo approach
- Update readme
- Update debug command
- Remove comment
- Add note on wokwi simulation
- Add the option to pull the image
- Move sh scripts into a folder and update paths
- Update devcontainers question
- Format output file
- Update scripts to use project-name instead of crate-name
- Add python3-venv dependency which is now required for esp-idf master
- Update CI with devcontainer argument
- Rename gitpod files
- Fix 4.4 build
- Fix 4.4 build
- Fixed syntax error in the project creation command
- Fixed a syntax error in project creation command
- Update Cargo.toml
- Update Cargo.toml
- Improve devcontainer support (#34)
- Remove toolchain question
- Fix CI
- Add CI to build all the targets (#35)
- Update toolchain version and remover clear-cache argument
- Upgrade rust edition
- Update CI to use the branch that triggered the action
- Update conditions
- Pathc https://github.com/esp-rs/esp-idf-svc/pull/98
- PRs should allow forked repos
- Update repository link
- Restore Cargo.toml
- Fix typo
- Update toolchain version to `1.62.1.0`
- Update cargo-generate installation methods
- Cleanup
- Add `continue-on-error: true`
- Update repository
- Update toolchain version to `1.63.0.2`
- Update `editor.formatOnSaveMode` property
- Simplify CI
- Simplify CI
- Add `ldproxy` binary install
- Fix ldproxy path
- Fix typo
- Remove toolchain argument
- Remove toolchain argument
- Add toolchain argument
- ci: ⚡️ Add caching
- ci: 🎨 Fix typo
- chore: ⬆️ Upgrade toolchain version to 1.64.0.0
- chore: ⬆️ Update embuild and esp-idf-sys version
- feat: ⚡️ Add custom runner
- docs: 📝 Add prompts documentation
- chore: 📝 Move cargo-first documentation to its readme
- chore: 📝 Fix format
- chore: 📝 Update order
- .gitignore: fix absolute path references
- Add the 64bit time_t flag when targeting v5 (#65)
- Try specifying rustflags in a different way which is known to work
- #66
- Update dependencies
- Update dependencies
- Retire espmonitor from the docu
- Update ESP IDF master CI
- docs: 📝 Update flashing and building sections
- docs: 📝 Fix typos
- docs: 📝 Move CI badge
- docs: 📝 Update prerequisites
- Name the master CI properly
- docs: 📝 Add note on RISC-V targets
- docs: 📝 Update RISCV install instructions
- Fix comments and simplify vars
- Cleanup the instructions
- Reference to cargo-espflash don't help
- Update README.md
- Put all cargo subcommands at a single place
- espup should be installed before use
- Update README.md
- Update README-cmake.md
- Update README-cmake.md
- Update README.md
- Update README-pio.md
- Update README-cmake.md
- Remove anyhow as a build dependency. It did not provide much value (#73)
- Update link_patches() comment to reflect current status (#72)
- Update README-cmake.md
- Project proposal
- rust prereqs
- move esp-idf-template files in preparation for merge
- remove vestigial files from ci-esp-idf-template
- move around template for airlock-loadout
- move and edit airlock-loadout template
